### PR TITLE
feat: support passing KIBANA_VERSION

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR adds a way to instrument the test framework to consume a different Docker image for Kibana: instead of using the `STACK_VERSION` variable, Kibana container version will be managed by `KIBANA_VERSION`. If it's not set, then the Docker image related to a Kibana PR will be used. This image is stored under the `observability-ci` Docker namespace.

We are also adding a pipeline that is listening for a webhook on the Kibana repo, and when the comment `/run-fleet-e2e-tests` is added to a PR by an Elastician, then the new pipeline will be triggered building and pushing the Docker image for that PR, and finally using that image in the e2e tests. This new pipeline will allow running it manually, passing the PR ID as an input parameter.

## Why is it important?
It will support the ability of running the tests against Kibana PRs (specially for Fleet), using the Dockerised version of Kibana for the PR.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

## How to test this PR locally

**Running a PR version of Kibana**, i.e. `pr94204`, that we already know it was pushed to the registry:
```shell
SUITE="fleet" TAGS="fleet_mode_agent && centos && enroll" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE KIBANA_VERSION=pr94204 BEATS_USE_CI_SNAPSHOTS=true DEVELOPER_MODE=true make -C e2e functional-test
```
>Please pay attention to the `BEATS_USE_CI_SNAPSHOTS=true` variable. 

**Running default version of Kibana**, i.e. `8.0.0-SNAPSHOT`:
```shell
SUITE="fleet" TAGS="fleet_mode_agent && centos && enroll" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true make -C e2e functional-test
```

## Related issues

- Relates #901

## Follow-ups
We'd need to add a Jenkins pipeline listening for Kibana PRs, with the proposed **synchronous** workflow (@EricDavisX @ph, please confirm):

1. a PR is sent to Kibana
   a. the author comments with `/run-fleet-e2e-tests`, or
   b. the modified paths affect Fleet code
1. the Kibana Docker image for that PR is built by the existing feature of `Deploy my Kibana`, without deploying it to the test cluster
1. The e2e tests in this repo are triggered, using `KIBANA_VERSION=prXYZ`
1. After the tests run, a commit status is added with the test results.

### CODE REVIEW

**Review by EricDavisX — APPROVED** [review]
With a brief review, it LGTM;

**Comment by EricDavisX** [comment]
Hi also, with regards to:
1a. the author comments with /run-fleet-e2e-tests...

I wanted to acknowledge it would be helpful to have an explicit, and easily edited list of the e2e 'suites' that are included when we make the test run.  Is that implied or otherwise easily done?  I didn't see (maybe reading too fast) where in the current changes that we could control the list.  For instance, do we want it to run the separate Helm chart or other Beats tests?  Maybe.  Maybe not, it would be good to make it explicit.

**Comment by mdelapenya** [comment]
> Hi also, with regards to:
> 1a. the author comments with /run-fleet-e2e-tests...
> 
> I wanted to acknowledge it would be helpful to have an explicit, and easily edited list of the e2e 'suites' that are included when we make the test run. Is that implied or otherwise easily done? I didn't see (maybe reading too fast) where in the current changes that we could control the list. For instance, do we want it to run the separate Helm chart or other Beats tests? Maybe. Maybe not, it would be good to make it explicit.

This PR is adding the support, not changing the existing behavior, as KIBANA_VERSION would be the same as the stack.

The second part will be a jenkins pipeline listening for kibana events on PRs, as the comment I suggested. This is not yet I'm place.

About the different tests suites, the new pipeline will/should trigger the fleet tests, only. Exactly the same as the nightly build does.

**Review by cachedout — APPROVED** [review]

File: .ci/e2eKibana.groovy [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  }
+  stages {
+    stage('Process GitHub Event') {
+      when {
+        beforeAgent true
+        anyOf {
+          expression { env.action == "edited" }
+          expression { env.action == "opened" }
+          expression { env.action == "reopened" }
+          expression { env.action == "synchronize" }
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:56]: We need to create a new expression to capture GH comments: i.e. `/run-fleet-e2e-tests`

File: .ci/e2eKibana.groovy [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      silentResponse: true,
+      regexpFilterText: "$pr_head_ref-$x_github_event",
+      regexpFilterExpression: '^(refs/tags/current|refs/heads/master/.+)-comment$'
+    )
+  }
+  stages {
+    stage('Process GitHub Event') {
+      when {
+        beforeAgent true
+        expression { env.repo_name == 'kibana' }
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:52]: Is it possible to listen to kibana events, only?

File: .ci/e2eKibana.groovy [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        [key: 'action', value: '$.action'],
+        [key: 'pr_id', value: '$.number'],
+        [key: 'pr_base_ref', value: '$.pull_request.base.ref'],
+        [key: 'pr_head_ref', value: '$.pull_request.head.ref'],
+        [key: 'pr_head_sha', value: '$.pull_request.head.sha'],
+        [key: 'pr_title', value: '$.pull_request.title'],
+        [key: 'repo_name', value: '$.repository.name'],
+      ],
+      genericHeaderVariables: [
+        [key: 'x-github-event', regexpFilter: '']
```
Comment (kuisathaverat) [inline .ci/e2eKibana.groovy:37]: you can use the `regexpFilter` to filter the actions

File: .ci/jobs/kibana-e2e-tests-mbp.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (kuisathaverat) [inline .ci/jobs/kibana-e2e-tests-mbp.yml:1]: it is a pipeline so the suffix `-mbp` can be removed

**Review by kuisathaverat — APPROVED** [review]

File: .ci/e2eKibana.groovy [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+     ],
+    genericHeaderVariables: [
+     [key: 'x-github-event', regexpFilter: 'comment']
+    ],
+     causeString: 'Triggered on comment: $GT_BODY',
+     printContributedVariables: false,
+     printPostContent: false,
+     silentResponse: true,
+     regexpFilterText: '$GT_REPO$GT_BODY',
+     regexpFilterExpression: '^elastic/kibana/run-fleet-e2e-tests$'
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:42]: The command to run the tests in Kibana would be `run-fleet-e2e-tests`

File: .ci/e2eKibana.groovy [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    )
+  }
+  parameters {
+    string(name: 'kibana_sha', defaultValue: "", description: "Commit to be used to run the Fleet E2E Tests. (e.g 1234567890abcdef)")
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:46]: Support running the tests from the UI

File: .ci/e2eKibana.groovy [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+  def parameters = [
+    booleanParam(name: 'forceSkipGitChecks', value: true),
+    booleanParam(name: 'forceSkipPresubmit', value: true),
+    booleanParam(name: 'notifyOnGreenBuilds', value: !isPR()),
+    booleanParam(name: 'BEATS_USE_CI_SNAPSHOTS', value: true),
+    string(name: 'runTestsSuites', value: suite),
+    string(name: 'GITHUB_CHECK_NAME', value: env.GITHUB_CHECK_E2E_TESTS_NAME),
+    string(name: 'GITHUB_CHECK_REPO', value: env.REPO),
+    string(name: 'GITHUB_CHECK_SHA1', value: getCommit()),
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:105]: The commit to be used is passed here, using https://github.com/elastic/e2e-testing/pull/905/files#diff-ffe928572b6fdb91893a16cfc20f4b449acb6733b00a2051bf59b2e4c51872f8R72-R78

File: .ci/e2eKibana.groovy [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+  log(level: 'DEBUG', text: "Triggering '${suite}' E2E tests for PR-${env.GT_PR}.")
+
+  // Kibana's maintenance branches follow the 7.11, 7.12 schema.
+  def branchName = "${env.GT_BASE_REF}.x"
+  def e2eTestsPipeline = "e2e-tests/e2e-testing-mbp/${branchName}"
+
+  def parameters = [
+    booleanParam(name: 'forceSkipGitChecks', value: true),
+    booleanParam(name: 'forceSkipPresubmit', value: true),
+    booleanParam(name: 'notifyOnGreenBuilds', value: !isPR()),
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:100]: I wonder if this is correct. We can turn it notifications off
```suggestion
    booleanParam(name: 'notifyOnGreenBuilds', value: false),
```

File: .ci/e2eKibana.groovy (hunk 5 above)
Comment (mdelapenya) [reply .ci/e2eKibana.groovy:42]: @EricDavisX this pipeline represents the way to interact with Kibana PRs. Will document it, but:

- In a kibana PR, add a comment with `/run-fleet-e2e-tests`. Only for elasticians
- Using the UI: trigger the pipeline using as the only parameter the GIT SHA of the kibana PR. (_We still have to capture the version of kibana for that commit SHA_)

File: .ci/e2eKibana.groovy [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+     causeString: 'Triggered on comment: $GT_BODY',
+     printContributedVariables: false,
+     printPostContent: false,
+     silentResponse: true,
+     regexpFilterText: '$GT_REPO$GT_BODY',
+     regexpFilterExpression: '^elastic/kibana/run-fleet-e2e-tests$'
+    )
+  }
+  parameters {
+    string(name: 'kibana_branch', defaultValue: "master", description: "Branch/PR to use to build the Docker image. (e.g 10000)")
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:46]: We are simplifying here the inputs: just use the PR ID, without `PR/` suffix

File: .ci/e2eKibana.groovy [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+  return "PR/${params.kibana_pr}"
+}
+
+def getDockerTag(){
+  if(env.GT_PR){
+    return "${env.GT_PR_HEAD_SHA}"
+  }
+
+  // we are going to use the 'pr12345' tag
+  return "pr${params.kibana_pr}"
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:87]: The Docker tag generated by the kibanaDockerBuild step builds and pushes the `pr12345` tag

File: .ci/e2eKibana.groovy [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+def hasCommentAuthorWritePermissions(prId, commentId){
```
Comment (mdelapenya) [inline .ci/e2eKibana.groovy:90]: Potentially reuse https://github.com/elastic/apm-pipeline-library/pull/1064

File: .ci/jobs/kibana-e2e-tests.yml [hunk 12]
```diff
@@ -0,0 +1,20 @@
+---
+- job:
+    name: e2e-tests/e2e-testing-kibana-fleet-mbp
```
Comment (mdelapenya) [inline .ci/jobs/kibana-e2e-tests.yml:3]:
```suggestion
    name: e2e-tests/e2e-testing-kibana-fleet
```

**Comment by mdelapenya** [comment]
I'm gonna merge this one, so that we can start using the new pipeline and detecting possible errors the soonest

**Pull request merged**